### PR TITLE
Set up Releases to work

### PR DESCRIPTION
**Pull request by LegacyKing**

### CODE REVIEW

**Comment by LegacyKing** [comment]
Sure

**Comment by karianna** [comment]
@LegacyKing merge conflicts (although this is still gated by 4743 anyhow)

**Comment by LegacyKing** [comment]
In light of the issues with #4743  I'd like to move forward on this without it being gated, as the goal is to remove releases from being just single dev involved (running 8+ hours these days).

File: .travis.yml [hunk 1]
```diff
@@ -5,51 +5,52 @@ dist: xenial
...
+- sudo apt-get install lib32z1 lib32ncurses5 nsis
+- export PERL_MM_USE_DEFAULT=1
+- chmod +x gradlew
     - if [[ $TRAVIS_OS_NAME == 'linux' ]]; then sudo apt-get install scons zlib1g-dev libcppunit-dev; fi
-    - if [[ $TRAVIS_OS_NAME == 'osx' ]]; then brew update; fi
-    - if [[ $TRAVIS_OS_NAME == 'osx' ]]; then brew install https://sourceforge.net/p/nsis/code/HEAD/tree/web-scripts/trunk/cppunit.rb?format=raw; fi
-    - if [[ $TRAVIS_OS_NAME == 'osx' ]]; then brew install scons mingw-w64 zlib; fi
- 
+- if [[ $TRAVIS_OS_NAME == 'osx' ]]; then brew update; fi
+- if [[ $TRAVIS_OS_NAME == 'osx' ]]; then brew install https://sourceforge.net/p/nsis/code/HEAD/tree/web-scripts/trunk/cppunit.rb?format=raw;
```
Comment (karianna) [inline .travis.yml]: Could have logic in one block

File: .travis.yml [hunk 2]
```diff
@@ -5,51 +5,52 @@ dist: xenial
...
+  fi
+- if [[ $TRAVIS_OS_NAME == 'osx' ]]; then brew install scons mingw-w64 zlib; fi
 install:
-    - wget -O /tmp/zlib.zip "https://downloads.sourceforge.net/project/libpng/zlib/1.2.8/zlib128-dll.zip?r=&ts=1403402496&use_mirror=hivelocity"
-    - mkdir /tmp/zlib
-    - unzip /tmp/zlib.zip -d /tmp/zlib
- 
+- wget -O /tmp/zlib.zip "https://downloads.sourceforge.net/project/libpng/zlib/1.2.8/zlib128-dll.zip?r=&ts=1403402496&use_mirror=hivelocity"
```
Comment (karianna) [inline .travis.yml]: I'd recommend extracting the version to a variable at the top of the script

File: .travis.yml [hunk 3]
```diff
@@ -5,51 +5,52 @@ dist: xenial
...
-    - ./gradlew pcgenRelease
+- "./gradlew pcgenRelease"
 
 deploy:
-    provider: releases
-    #Replace this with a real token
-    api_key: "GITHUB OAUTH TOKEN"
+  provider: releases
+  api_key:
+    secure: vcWcTepSQfgZTbLrC2jbJKLFB8pho72dz00vqUyWjtilvLcbOYVgVWgE5U5zjPV+VS5j4cj93sMWv74dpwamXGv/kIld7fQlpVkMELtK5AkRa2bjOMi0BHk54K8gcGOG//Gen/3gdBJfV1kYcsOhC2Rc4t4pcY2rkW534NLUxdg=
```
Comment (karianna) [inline .travis.yml]: Are we leaking a secret here?

File: .travis.yml [hunk 4]
```diff
@@ -5,51 +5,52 @@ dist: xenial
...
+    file: build/release/*
     skip_cleanup: true
-    on:
+  on:
+    repo: PCGen/pcgen
         #This make the deploy only happen when a tag is created.  
         tags: true
         #Only on master
-        branch: master
+        #branch: master
```
Comment (karianna) [inline .travis.yml]: commented out code is usually a no-no

File: .travis.yml (hunk 1 above)
Comment (LegacyKing) [reply .travis.yml]: This is the code the Travis team has...

File: .travis.yml (hunk 3 above)
Comment (LegacyKing) [reply .travis.yml]: It's how the travis set this up. That was a script add.

**Comment by grimreaper** [comment]
I release my hold on this PR. Will fix my other ones later.

**Comment by karianna** [comment]
@LegacyKing - Looks like a conflict

**Comment by LegacyKing** [comment]
Use the Squash Option.

**Pull request merged**